### PR TITLE
[bitnami/jenkins] Major release 12.0.0

### LINKED ISSUE #14496: unable to scale jenkins pod
### Name and Version

bitnami/jenkins, latest

### What steps will reproduce the bug?

`helm install jenkins jenkins`

### What is the expected behavior?

By default, it only creates 1 pod; but I want to increase no of pods so that I can achieve master-slave architecture. But can not find any way to do that.

### What do you see instead?

No options for scaling are provided

### PR DESCRIPTION
**Pull request by migruiz4**
Signed-off-by: Miguel Ruiz <miruiz@vmware.com>

### Description of the change

This major release includes several changes:
- Adds support for new plugin features.
- Add support for both initScripts (docker-entrypoint-initdb.d - Executed during first boot) and initHookScripts (init.groovy.d - Executed on every restart).
- Users can now provide their own JKS keystore or automatically generate one with self-signed certificates.
- Implement features for new supported plugins 'configuration-as-code' and 'kubernetes'.
- Add RBAC features.

- Fixes #14496

### Possible drawbacks

Jenkins deployment adds a new component label, which requires a major release.

New features that involve installing additional plugins will be tagged as experimental during this major release but may become the default in a future major release.

Enabling them on previously existing deployments may require overriding plugins and existing configuration, please be cautious and create backups beforehand. 

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by migruiz4** [comment]
Please note the CI/CD tests will fail until the Jenkins image is updated with the new features

**Review by agomezmoron — DISMISSED** [review]
LGTM

File: bitnami/jenkins/templates/role.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    verbs:
+      - "get"
+      - "watch"
+      - "list"
+  {{- end }}
+  {{- if .Values.rbac.rules }}
+  {{- include "common.tplvalues.render" ( dict "value" .Values.rbac.rules "context" $ ) | nindent 2 }}
+  {{- end }}
+{{- end }}
+
```
Comment (carrodher) [inline bitnami/jenkins/templates/role.yaml:55]:
```suggestion
```

**Comment by migruiz4** [comment]
These changes have been tested using the following `values.yaml`:
```yaml
plugins:
  - kubernetes:3852.v41ea_166a_ed1b_
  - workflow-aggregator:590.v6a_d052e5a_a_b_5
  - git:4.13.0
  - configuration-as-code:1569.vb_72405b_80249
configAsCode:
  enabled: true
agent:
  enabled: true
```

The Kubernetes agent and configAsCode functionalities have been tested doing the following:
* Login into Jenkins > Create a 'Freestyle Project' > Add the label 'kubernetes-agent' and a sample shell command > Save project and run
* Attempts to modify configuration through the UI will be overwritten if `configAsCode.autoReload.enabled=true` is set, changes performed by editing the ConfigMap will trigger an automatic config reload.

**Review by agomezmoron — APPROVED** [review]
LGTM, great work @migruiz4

**Pull request merged**